### PR TITLE
ATO-981: Send sessionId to Auth `/userinfo` endpoint 

### PR DESCRIPTION
**Pull request by Ryan-Andrews99**
## What
- Sends the session id in the headers of the `/userinfo` request from orchestration to auth
- Adds some logging to log this sessionId if present in the headers of the request

## How to review
- Code review commit-by-commit 
- Tested by sequentially deploying in Orch dev, running through a journey and then deploying auth sandpit and running through again, all went fine!
## Checklist

- [x] Impact on orch and auth mutual dependencies has been checked.

- [ ] A UCD review has been performed.

### CODE REVIEW

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java [hunk 1]
```diff
@@ -318,15 +318,15 @@ public APIGatewayProxyResponseEvent handleRequest(
                         buildURI(
                                 configurationService.getAuthenticationBackendURI().toString(),
                                 "userinfo");
-
+                var sessionId = userSession.getSessionId();
                 HTTPRequest authorizationRequest = new HTTPRequest(GET, userInfoURI);
+                authorizationRequest.setHeader("session-id", sessionId);
```
Comment (billrdunn) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java:323]: I think that typically header keys are capitalised, ie. `Session-Id`

**Review by billrdunn — DISMISSED** [review]

**Review by billrdunn — DISMISSED** [review]
Approved subject to testing in dev

**Review by billrdunn — APPROVED** [review]

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java [hunk 2]
```diff
@@ -79,6 +79,7 @@
 import static uk.gov.di.authentication.oidc.domain.OrchestrationAuditableEvent.AUTH_UNSUCCESSFUL_USERINFO_RESPONSE_RECEIVED;
 import static uk.gov.di.orchestration.shared.conditions.DocAppUserHelper.isDocCheckingAppUserWithSubjectId;
 import static uk.gov.di.orchestration.shared.conditions.IdentityHelper.identityRequired;
+import static uk.gov.di.orchestration.shared.domain.RequestHeaders.SESSION_ID_HEADER;
```
Comment (dbes-gds) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java:82]: importing orch classes into the auth dirs?

Comment (Ryan-Andrews99) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java:82]: Good spot, this was meant to be the Auth ones 🤦

Comment (Ryan-Andrews99) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java:82]: Ah wait is this not an Orchestration lambda and should use the shared orchestration code?

Comment (dbes-gds) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthenticationCallbackHandler.java:82]: yes you're right thanks :)

**Review by dbes-gds — APPROVED** [review]

**Pull request merged**